### ISSUE TITLE
Drupalgap is not working in offline mode.

### ISSUE THREAD

**Issue by zawlinoo** [signalpoint/DrupalGap#207]
Can't access the app when its offline mode. Am I wrong inside the setting or not supported yet?

**Comment by signalpoint** [comment]
@zawlinoo Chances are offline mode is buggy. Will you please describe "can't access the app" in more detail please?

**Comment by zawlinoo** [comment]
Sorry.. I was saying that i can't  get into the app interface. Only white dead screen when I clicked my app in offline mode.

**Comment by signalpoint** [comment]
I just realized there is a missing plugin from the documentation:

`
cordova plugin add org.apache.cordova.dialogs
`

Add that and the white screen of death should go away.

**Comment by zawlinoo** [comment]
I was adding this org.apache.cordova.dialogs plugin and its show not white screen, Its appear Fail Connection. Thanks. But I'm not sure, it would be better we can see nodes and other interface in offline mode.
Thanks for your solution.

**Comment by signalpoint** [comment]
@zawlinoo I agree, offline mode should go directly to the app's interface. Perhaps with a simple warning message saying "Offline mode". I'll work on improving that.

As for being able to see nodes while in offline mode, your app will have to dynamically retrieve the required nodes (and cache them, see settings.js) during hook_install(). Alternatively, you'll have to hard code the JSON object of the node (http://www.example.com/?q=drupalgap/node/123.json) and save it to local storage using the proper key:

http://api.drupalgap.org/global.html#entity_local_storage_key

**Comment by zawlinoo** [comment]
Thank a lots for your all hard works. I'll try on Monday and I'll tell you about that.

**Comment by zawlinoo** [comment]
this is my function in my custom module 

```
function custom_module_local_storage_key(entity_type, custom_module_article_list) {
  try {
    return entity_type + '_' + custom_module_article_list;
  }
  catch (error) { drupalgap_error(error); }
}
```

still the same. no cache yet.  id parameter should be item list populated id right? this is my id custom_module_article_list but still getting offline mode. What will entity_type should be? Sorry, I'm not expert.

**Comment by signalpoint** [comment]
Sorry, I don't understand what you are trying to do.

FYI, there is no hook_local_storage_key().

Stop by #drupal-drupalgap on IRC to chat about this in more detail, and it helps to share your code in your own repository on github.

**Comment by zawlinoo** [comment]
Sorry, I was login to IRC chat no reply yet. I was push my dev version to my repo. Please have a look and i was adding pull refresh for node list also. Kind of messy now. Here is my repo. 

https://github.com/zawlinoo/drupalgapdev

**Comment by signalpoint** [comment]
https://drupal.org/irc

**Comment by bircher** [comment]
I am interested in the offline mode too. the first thing for me to do was to allow offline simulation with ripple.
https://github.com/bircher/DrupalGap/commit/3cdc2fabeb5fa2c1fa15c218380a4ba822317503
The problem describes on stackoverflow persists as such. The app doesn't switch to online mode even if the ripple setting was changed.
But at least one can work offline.

**Comment by signalpoint** [comment]
This issue has been closed in favor of this issue: #258
